File: python/problem010.py
import math

number_array = [True]*2000000

for i in range(2, int(math.sqrt(2000000))):
    if number_array[i]:
        j = i+i
        while j < 2000000:
            number_array[j] = False
            j += i

result = 0
for i in range(2, 2000000):
    if number_array[i]:
        result += i
print(result)

File: java/src/Problem010.java
import java.math.BigInteger;
import java.util.Arrays;

/**
 * Created by 김 on 2016-02-05.
 */
public class Problem010 {

    boolean numberArray[] = new boolean[2000000];

    void set() {
        Arrays.fill(numberArray, true);
        for (int i = 2; i < Math.sqrt(2000000); i++) {
            if (numberArray[i]) {
                for (int j = i + i; j < 2000000; j = j + i) {
                    numberArray[j] = false;
                }
            }
        }
    }

    BigInteger result(){
        set();
        BigInteger value = new BigInteger("0");
        for(int i=2; i<2000000; i++){
            if(numberArray[i]){
                value = value.add(BigInteger.valueOf(i));
            }
        }
        return value;
    }

    public static void main(String args[]) {
        Problem010 problem = new Problem010();
        System.out.println(problem.result().toString());
    }
}
